### PR TITLE
Homework3: Shevchuk Aleksei

### PR DESCRIPTION
**Pull request by alex-shef**

### CODE REVIEW

File: Homework3/task1.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# 1. FizzBuzz
+# Напишите программу, которая печатает цифры от 1 до 100,
+# но вместо чисел, кратных 3 пишет Fizz, вместо чисел кратный 5 пишет Buzz,
+# а вместо чисел одновременно кратных и 3 и 5 - FizzBuzz.
+
+lst = [str(num) for num in range(1, 101)]
+for num in range(len(lst)):
+    if int(lst[num]) % 3 == 0 and int(lst[num]) % 5 == 0:
```
Comment (belovmd) [inline Homework3/task1.py]: 15?

File: Homework3/task1.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# 1. FizzBuzz
+# Напишите программу, которая печатает цифры от 1 до 100,
+# но вместо чисел, кратных 3 пишет Fizz, вместо чисел кратный 5 пишет Buzz,
+# а вместо чисел одновременно кратных и 3 и 5 - FizzBuzz.
+
+lst = [str(num) for num in range(1, 101)]
```
Comment (belovmd) [inline Homework3/task1.py]: seems to be redundant code

File: Homework3/task3.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+tpl1 = tuple(lst1)
+print(tpl1)
+
+# Создайте кортеж ('a', 'b', 'c'), И сделайте из него список.
+tpl2 = ('a', 'b', 'c')
+lst2 = list(tpl2)
+print(lst2)
+
+# Сделайте следующие присвоения одной строкой a = 'a', b=2, c=’python’.
+lst2 = ['a', 2, 'python']
```
Comment (belovmd) [inline Homework3/task3.py]: it doesn't create variables a, b, c

File: Homework3/task5.py [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+# Дан список.
+# Выведите те его элементы, которые встречаются в списке только один раз.
+# Элементы нужно выводить в том порядке, в котором они встречаются в списке.
+
+lst = [int(num) for num in input('Enter string of numbers: ').split()]
+result_list = []
+for elem in lst:
+    if lst.count(elem) == 1:
+        result_list.append(elem)
+print(' '.join([str(num) for num in result_list]))
```
Comment (belovmd) [inline Homework3/task5.py]: print(num, end=' ')

File: Homework3/task7 additionally.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+else:
+    print('1.NO')
+
+# 2.Определите, можно ли отломить от шоколадки ровно k одиночных долек.
+if (k == 2 and (m, n == 1, 2 or n, m == 1, 2)) or\
+        (k == 1 and (m == 1 or n == 1)):
+    print('2.YES')
+else:
+    print('2.NO')
+# Т. к. ломаем только один раз по одной линиии, могут быть только 2 случая,
```
Comment (belovmd) [inline Homework3/task7 additionally.py:20]: we can cut any times we need

File: Homework3/task7 additionally.py [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+# когда m или n равно 1.
+
+# 3.Определите, можно ли отломить от шоколадки ровно k одиночных долек
+# с помощью m разломов.
+if k == m or (n * m >= k and m > n and k % n == 0) or (k, m, n == 2, 1, 2):
+    print('3.YES')
+else:
+    print('3.NO')
+# Решение справедливо при допустимости кол-ва разломов не больше m.
```
Comment (belovmd) [inline Homework3/task7 additionally.py]: it's possible sometimes when t>m

**Review by belovmd — COMMENTED** [review]
please address these issues

File: Homework3/task1.py (hunk 1 above)
Comment (alex-shef) [reply Homework3/task1.py]: instead of 15 gives "FizzBuzz"

File: Homework3/task7 additionally.py (hunk 5 above)
Comment (alex-shef) [reply Homework3/task7 additionally.py:20]: but then the number of required slices "k" should just be less than NxM

Comment (alex-shef) [reply Homework3/task7 additionally.py:20]: and "k" must be a multiple of "N" or "M"

File: Homework3/task1.py [hunk 7]
```diff
@@ -0,0 +1,21 @@
+# 1. FizzBuzz
+# Напишите программу, которая печатает цифры от 1 до 100,
+# но вместо чисел, кратных 3 пишет Fizz, вместо чисел кратный 5 пишет Buzz,
+# а вместо чисел одновременно кратных и 3 и 5 - FizzBuzz.
+
+lst = []
+for num in range(1, 101):
+    if num % 3 == 0 and num % 5 == 0:
```
Comment (belovmd) [inline Homework3/task1.py:8]: if not num % 15 looks much more simple

**Review by belovmd — COMMENTED** [review]
9

**Pull request closed without merging**